front/src/applications/operationalStudies/views/Scenario/components/ManageTimetableItem/Itinerary/hooks/usePathStepsMetadata.ts

```diff
@@ -0,0 +1,247 @@
+  const { infraId, getTrackSectionsByIds } = useScenarioContext();
+
+  const [pathStepsMetadataById, setPathStepsMetadataById] = useState<Map<string, PathStepMetadata>>(
+    new Map()
+  );
+
+  // 1. Extract the train path to extract its steps related operational points
+  const strippedTrainPath: TrainSchedule['path'] = useMemo(
+    () =>
+      pathSteps.reduce<TrainSchedule['path']>((acc, step) => {
+        if (!step.location) return acc;
+        if ('uic' in step.location || 'trigram' in step.location) {
+          // strip location from its secondary_code so we can have all matchs for an uic or trigram
+          const {
+            secondary_code: _secCode,
+            track_reference: _trackRef,
+            ...location
+          } = step.location;
+          acc.push({
+            id: step.id,
+            ...location,
+          });
+          return acc;
+        }
+        acc.push({
+          id: step.id,
+          ...step.location,
+        });
+        return acc;
+      }, []),
+    [pathSteps]
+  );
+
+  const pathStepsOperationalPoints = usePathOps(infraId, strippedTrainPath, {
+    returnAllOps: true,
+  });
+
+  // 2. Since a path step containing 'operational_point' as location will have only one match
+  // with postInfraByInfraIdMatchOperationalPoints, we need to call the endpoint again with
+  // the opId corresponding uic in order to get all the possible matches and have
+  // all the secondary codes and track names
+  const uicPayload: Extract<OperationalPointReference, { uic: number }>[] = useMemo(() => {
+    const opIds = pathSteps.reduce<string[]>((acc, step) => {
+      if (step.location && 'operational_point' in step.location) {
+        acc.push(step.location.operational_point);
+      }
+      return acc;
+    }, []);
+
+    if (opIds.length === 0) return [];
+
+    return pathStepsOperationalPoints.reduce<{ uic: number }[]>((acc, op) => {
+      const uic = op.extensions?.identifier?.uic;
+      if (uic && opIds.includes(op.id)) {
+        acc.push({ uic });
+      }
+      return acc;
+    }, []);
+  }, [pathSteps, pathStepsOperationalPoints]);
+
+  const { currentData: allOpIdsOperationalPoints } =
+    osrdEditoastApi.endpoints.postInfraByInfraIdMatchOperationalPoints.useQuery(
+      uicPayload.length > 0
+        ? {
+            infraId,
+            body: {
+              operational_point_references: uicPayload,
+            },
+          }
+        : skipToken
+    );
+
+  // 3. Merge both operational points lists to have all possible matches
+  const allOps = useMemo(() => {
+    if (
+      !allOpIdsOperationalPoints?.related_operational_points ||
+      allOpIdsOperationalPoints.related_operational_points.length === 0
+    ) {
+      return pathStepsOperationalPoints;
+    }
+    return [
+      ...pathStepsOperationalPoints,
+      ...allOpIdsOperationalPoints.related_operational_points.flat(),
+    ];
+  }, [pathStepsOperationalPoints, allOpIdsOperationalPoints]);
```

Comment:
I think this works too :thinking: 
```suggestion
  const allOps = useMemo(
    () => [
      ...pathStepsOperationalPoints,
      ...(allOpIdsOperationalPoints?.related_operational_points || []).flat(),
    ],
    [pathStepsOperationalPoints, allOpIdsOperationalPoints]
  );
```

Reply:
Done